src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -77,6 +79,24 @@ private QueryPhaseSearcher getQueryPhaseSearcher(final SearchContext searchConte
 
     @VisibleForTesting
     protected Query extractHybridQuery(final SearchContext searchContext, final Query query) {
+        if (isHybridQueryExtendedWithDlsRules(query, searchContext)) {
```

Comment:
most of the new code in this method is almost the same as in HybridCollectorManager. Can you make that common code sharable. Easiest way is to create a new method in HybridQueryUtil (name can be extractHybridQuery), is can handle all 4 cases of query wrapping (DLS, nested DLS, standard boolean, plain hybrid).

Reply:
Done, the shared part has been moved to HybridQueryUtil